### PR TITLE
TCP proxy for tests and cancel tasks for SimpleServer

### PR DESCRIPTION
**Pull request by mosquito**

### CODE REVIEW

File: aiomisc_pytest/pytest_plugin.py [hunk 1]
```diff
@@ -27,6 +29,182 @@
...
+
+        self.loop = asyncio.get_event_loop()
+        self.client_reader = client_reader  # type: asyncio.StreamReader
+        self.client_writer = client_writer  # type: asyncio.StreamWriter
+
+        self.server_reader = None  # type: t.Optional[asyncio.StreamReader]
+        self.server_writer = None  # type: t.Optional[asyncio.StreamWriter]
+
+        self.tasks = ()                     # type: t.Iterable[asyncio.Task]
+        self.chunk_size = chunk_size        # type: int
```
Comment (dizballanze) [inline aiomisc_pytest/pytest_plugin.py:58]: I'd rather remove comments indention aligning. It's not convenient to realign after changes.

File: aiomisc_pytest/pytest_plugin.py [hunk 2]
```diff
@@ -27,6 +29,182 @@
...
+                data = await reader.read(self.chunk_size)
+                processor_func = getattr(self, processor)
+
+                if not data:
+                    return
+
+                if processor_func:
+                    data = await aiomisc.awaitable(processor_func)(data)
+
+                if self.delay > 0:
```
Comment (dizballanze) [inline aiomisc_pytest/pytest_plugin.py:80]: `if self.delay:`

File: aiomisc_pytest/pytest_plugin.py [hunk 3]
```diff
@@ -27,6 +29,182 @@
...
+                    self.server_reader,
+                    self.client_writer,
+                    "write_processor",
+                ),
+            ),
+            self.loop.create_task(
+                self.pipe(
+                    self.client_reader,
+                    self.server_writer,
+                    "read_processor",
```
Comment (dizballanze) [inline aiomisc_pytest/pytest_plugin.py:109]: ?

File: aiomisc_pytest/pytest_plugin.py [hunk 4]
```diff
@@ -27,6 +29,182 @@
...
+        self, target_host: str, target_port: int,
+        listen_host: str = "127.0.0.1",
+    ):
+        self.target_port = target_port
+        self.target_host = target_host
+        self.proxy_host = listen_host
+        self.proxy_port = get_unused_port()
+        self.delay = 0
+
+        self.clients = set()        # type: t.Set[TCPProxyClient]
```
Comment (dizballanze) [inline aiomisc_pytest/pytest_plugin.py:140]: There should be 2 spaces before comment

File: tests/pytest_plugin/test_tcp_proxy.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+@pytest.fixture()
+@async_generator
+async def proxy(tcp_proxy, localhost, server_port):
+    proxy = tcp_proxy(localhost, server_port)
+
+    try:
+        await yield_(proxy)
+    finally:
+        await asyncio.wait_for(proxy.close(), timeout=1)
```
Comment (dizballanze) [inline tests/pytest_plugin/test_tcp_proxy.py:55]: Why do you need custom timeouts here?

File: aiomisc_pytest/pytest_plugin.py (hunk 2 above)
Comment (mosquito) [reply aiomisc_pytest/pytest_plugin.py:79]: No.

**Pull request merged**